### ISSUE TITLE
How to avoid hard coding prebuilds folder's path 

### ISSUE THREAD

**Issue by qmas** [Huddly/sdk#66]
I have created an app.exe using pkg. The app.exe depends on Huddly SDK. The app.exe runs fine when I change the d:\HuddlyIQ\node_modules\node-gyp-build\index.js file with hard coded prebuilds path. Without hard coding prebuilds path, the node-gyp-build\index.js throws error message "Error: No native build was found for runtime=node abi=57 ...". 
How can I run app.exe without making any changes to SDK files or the  \node-gyp-build\index.js ?

**Comment by jontore** [comment]
It depends on how you build the exe file, but most of the time the binary dependencies has to be packed, relative to the exe file. So they need to be in ```node_modules/usb/prebuilds```. 

if you use https://github.com/zeit/pkg

you need to add this to your package.json file 

```
"pkg": {
    "assets": "node_modules/**/win32-x64/node-57.node"
}
```

**Comment by qmas** [comment]
Made the change you suggested. I have attached Package.json file that I am using to create exe package. Please see the highlighted error message produced when I run this exe file. 

![Package_json](https://user-images.githubusercontent.com/13786933/54150288-771b0080-440e-11e9-9b91-ab7b19d2c679.PNG)

**Comment by jontore** [comment]
Hmm, what version are you building this with.

It also look like you also do something before the build step have you tried to just run the pkg binary.

```node_modules/.bin/pkg .``` And you have to have .node files where ever you run the exe.

**Comment by qmas** [comment]
Thanks for the previous comment. My machine has pkg version is 4.36, Node version is 8.9.4.  I will read more about it. 

I can build  by the following command. 
`pkg .  -d --target win --output huddlyIQ.exe`
When I put exe and bin\*.node files in a same folder and hard code bin path in the node_modules\node-gyp-build\index.js file, then it runs fine. Without this change in Huddly SDK exe will not run. I am required to run exe without changing the SDK code.

**Comment by jontore** [comment]
Ok that sounds correct. No the .node files has to be in the same path as it would be in the node_modules folder
```
- huddlyIQ.exe
- node_modules
  - usb
    - prebuilds
      - win32-x64
        - node-57.node
```

**Comment by qmas** [comment]
I have put exe and .node files exactly as you have described in your previous comment. And the error message is:
 
`(node:7572) UnhandledPromiseRejectionWarning: Error: No native build was found for runtime=node abi=57 platform=win32glibc arch=x64
    at Function.load.path (D:\snapshot\HuddlyIQ\node_modules\node-gyp-build\index.js:62:9)
    at load (D:\snapshot\HuddlyIQ\node_modules\node-gyp-build\index.js:18:30)
    at Object.<anonymous> (D:\snapshot\HuddlyIQ\node_modules\usb\usb.js:1:125)
    at Module._compile (pkg/prelude/bootstrap.js:1254:22)
    at Object.Module._extensions..js (module.js:661:10)
    at Module.load (module.js:563:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Module.require (module.js:594:17)
    at Module.require (pkg/prelude/bootstrap.js:1159:31)
(node:7572) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7572) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process
with a non-zero exit code.`

Note: In VS Code, it runs fine. Just when I package then SDK paths get confused. If I hard code paths in SDK then it works too. I don't want any hard coding in SDK.

**Comment by jontore** [comment]
I get that, it's a bit hard to understand what goes wrong without seeing the code. Could you setup a minimal example where the problem is reproduced, could be problem with the version of pkg.

**Comment by qmas** [comment]
I trimmed the project down to an index.js file+HuddlySDK and created app.exe. This app.exe pulled camera information without an issue. That means path issue is elsewhere in my code that I have to debug. Thanks

You may close this issue.

**Comment by jontore** [comment]
I would recommend to add some logging in ```node_modules/node-gyp-build/index.js``` line 46, usually helps to see what binary it is looking for.

**Comment by qmas** [comment]
Just for a record: The issue was in folder name of my application. My app folder name had special characters '(' and ')'. When I renamed the folder by replacing these special characters with '_', the dependency issue got resolved. For example folder was renamed from "HuddlyIQ-(March 3)\" to HuddlyIQ_March_3\ and I don't see issue any more. When I renamed back to old name with special characters then issue reappeared. 
 `
